### PR TITLE
Update usaco-1188.mdx

### PR DESCRIPTION
**Pull request by Keys9873**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/platinum/usaco-1188.mdx [hunk 1]
```diff
@@ -144,4 +144,100 @@ int main() {
...
+```java
+import java.io.*;
+import java.util.*;
+
+public class MinimizingHaybales {
+	static int k;
+
+	public static class Block {
+		public int hi, lo;
+		ArrayList<Integer> vals;
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:157]:
```suggestion
		List<Integer> vals;
```

File: solutions/platinum/usaco-1188.mdx [hunk 2]
```diff
@@ -144,4 +144,100 @@ int main() {
...
+
+		public Block() {
+			vals = new ArrayList<>();
+			hi = (int)-1e9;
+			lo = (int)1e9;
+		}
+		public Block(ArrayList<Integer> a) {
+			vals = new ArrayList<>();
+			hi = (int)-1e9;
+			lo = (int)1e9;
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:167]: use integer.min_value/max_value

File: solutions/platinum/usaco-1188.mdx [hunk 3]
```diff
@@ -144,4 +144,100 @@ int main() {
...
+			boolean ans = vals.get(vals.size() - 1) == x;
+			if (test) { vals.remove(i); }
+			return ans;
+		}
+	}
+
+	public static void main(String[] args) throws Exception {
+		BufferedReader br =
+		    new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:203]:
```suggestion
```

File: solutions/platinum/usaco-1188.mdx [hunk 4]
```diff
@@ -144,4 +144,100 @@ int main() {
...
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int n = Integer.parseInt(st.nextToken());
+		k = Integer.parseInt(st.nextToken());
+		ArrayList<Block> sqrt = new ArrayList<>();
+		sqrt.add(new Block());
+		for (int i = 0; i < n; i++) {
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:209]: space out code

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by Keys9873** [comment]
Added comments too

File: solutions/platinum/usaco-1188.mdx [hunk 5]
```diff
@@ -143,5 +142,122 @@ int main() {
...
+			// the left
+			int j = sqrt.size() - 1;
+			while (j > 0 &&
+			       (x + k >= sqrt.get(j).hi && x - k <= sqrt.get(j).lo)) {
+				j--;
+			}
+
+			// Handle edge case if x doesn't cross an element greater than it in
+			// block j while swapping left Swaps element x to the right to the
+			// lexographically optimal position
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:231]: multiline comments

File: solutions/platinum/usaco-1188.mdx [hunk 6]
```diff
@@ -143,5 +142,122 @@ int main() {
...
+			if (j < sqrt.size() - 1 && sqrt.get(j).add(x, true)) {
+				j++;
+				while (j < sqrt.size() - 1 && x >= sqrt.get(j).hi) { j++; }
+			}
+			sqrt.get(j).add(x, false);
+
+			// If block is too large split into two smalller blocks of equal
+			// size
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:239]:
```suggestion
			// Split the block if it's too large
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/platinum/usaco-1188.mdx [hunk 7]
```diff
@@ -143,5 +142,122 @@ int main() {
...
+			if (sqrt.get(j).vals.size() > 300) {
```
Comment (SansPapyrus683) [inline solutions/platinum/usaco-1188.mdx:240]: 300?

Comment (Keys9873) [reply solutions/platinum/usaco-1188.mdx:239]: sqrt(1e5)

Comment (SansPapyrus683) [reply solutions/platinum/usaco-1188.mdx:247]: call Math.sqrt(n) or smth then

Comment (Keys9873) [reply solutions/platinum/usaco-1188.mdx]: i made it a constant

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
great!

**Pull request merged**